### ISSUE TITLE
Is the checksum check not commonly used?

### ISSUE THREAD

**Issue by chhs2131** [M0r13n/pyais#82]
Hi.
First of all, thank you for making a good library.

While using `decode()` in this library, I found that checksum check was not performed. (I proceeded with decoding by changing only the 'checksum' field.)
The output was printed by adding `print(nmea.is_vailed)` to `_assamble_messages()` belonging to `decode.py`. It seems that the result of `False` should be `raise`, but all decoding proceeds normally.

```python
# Mosaic messages for security reasons
is_vaild result:  True   ais_msg:  b'!AIVDM,1,1,,A,xxxxxxxxxxa3OsLEKwfmKB:L0@8g,0*55\n'
is_vaild result:  False   ais_msg:  b'!AIVDM,1,1,,A,xxxxxxxxxxa3OsLEKwfmKB:L0@8g,0*15\n'
is_vaild result:  False   ais_msg:  b'!AIVDM,1,1,,A,xxxxxxxxxxa3OsLEKwfmKB:L0@8g,0*10\n'
is_vaild result:  False   ais_msg:  b'!AIVDM,1,1,,A,xxxxxxxxxxa3OsLEKwfmKB:L0@8g,0*1A\n'
is_vaild result:  False   ais_msg:  b'!AIVDM,1,1,,A,xxxxxxxxxx3OsLEKwfmKB:L0@8g,0*B5'
```

Is the checksum check function not commonly used? Or is there another reason?
Thank you.   :)

**Comment by Inrixia** [comment]
Not throwing allows for still being able to work with decoded data even if the checksum fails. If you wanted to raise on a invalid checksum then you should be able to just include that check in your decoding pipeline.

**Comment by M0r13n** [comment]
Yes, @Inrixia is right. This library tends to be liberal in what it accepts. There are reasons for that:

This library is often used for data analysis. This means that a researcher analyzes large amounts of AIS messages. Such message streams might contain thousands of messages with invalid checksums. Its up to the researcher to decide whether he/she wants to include such messages in his/her analysis. Raising an exception for every invalid checksum would both cause a performance degradation because handling of such exceptions is expensive and make it impossible to include such messages into the analysis.

Therefore, we decided to not raise an exception. Instead its up to the user to decide what to do with invalid messages.

**Comment by Inrixia** [comment]
> This library is often used for data analysis.

Couldn't have said it better myself, this is also why this library supports returning `None` for properties that failed to decode allowing decoding of partial or incomplete messages. Which is a critical feature I have found as lots of AIS data I have worked with may have just the last bit of a message missing but otherwise be completely sound.

@chhs2131 If you have any further questions feel free to ask them. Either way we can close this issue now
